pkg/haproxy/config.go

```diff
@@ -222,6 +236,83 @@ type RenderConfiguration struct {
 
 	// The load balancer port. Is not currently configurable.
 	Port uint32
+
+	// ExtraServices is the additional services that loadbalancer proxies to
+	ExtraServices []Frontend
+}
+
+// Frontend represent the loadbalancer endpoint that will  loadbalance the request to the backend servers
+type Frontend struct {
+	// Frontend Name
+	Name string
+
+	// load balancer port
+	Port uint32
+
+	// backend that this front end loadbalances to
+	Backend *Backend
+}
+
+// Backend represent the backend servers that host a service
+type Backend struct {
+	// backend port
+	Port uint32
+
+	// Addresses of the machines for this backend
+	Addresses []corev1.EndpointAddress
+
+	// HealthCheckConfig for this backend
+	HealthCheckConfig *BackendHealthCheckConfig
+}
+
+type BackendHealthCheckConfig struct {
```

Comment:
Why do we need to redefine the health check types?

Reply:
I initially thought health check types here might have some different configs than the types in API that we are exposing. But, changed it now to use the same type  as it is good to be consistent